.github/actions/artifact_definition/action.yaml

```diff
@@ -0,0 +1,77 @@
+name: Artifact Definition
+description: Action to generate and upload an artifact
+
+# This action allows you to generate an artifact in a ZIP file and upload it.
+
+inputs:
+  workspace_name:
+    description: The name of the workspace to create the artifact for.
+    type: string
+    required: true
+  bundle_name:
+    description: Name of the bundle
+    type: string
+    required: false
+  base_workdir:
+    description: The base workdir (Default is ./out)
+    type: string
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      name: Checkout
+
+    - name: Build the artifact
+      id: make-artifact
+      shell: bash
+      run: |
+          set -e
+
+          # determine the entry point file name from package.json main property
+          ENTRY_POINT=$(jq -r '.main' package.json)
+
+          if [ -z $ENTRY_POINT ]; then
+            echo "::error::invalid target"
+            exit 1
+          fi
+
+          # generate node_modules folder excluding devDependencies
+          yarn workspaces focus --production
+
+          FORMAT=$(jq -r 'if .type == "module" then "esm" else "cjs" end' package.json)
+          SHORT_SHA=$(git rev-parse --short ${{ github.sha }})
+
+          mkdir ${{ github.run_id }}
+
+          # bundle compiled code, excluding node_modules 
+          curl -fsSL https://esbuild.github.io/dl/v0.21.5 | sh
+          ./esbuild index=$ENTRY_POINT --bundle --format=$FORMAT --platform=node --target=node20 --packages=external --outdir=${{ github.run_id }} --allow-overwrite
```

Comment:
are you sure we want to bulde code using `esbuild` before deploying this to a cdn?
usually web frameworks like NextJS use their embedded bundling method

Reply:
ditto